fiftyone/core/annotation/generate_label_schema.py

```diff
@@ -0,0 +1,569 @@
+"""
+Annotation label schema generation
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+import eta.core.utils as etau
+
+import fiftyone.core.annotation.constants as foac
+from fiftyone.core.annotation.validate_label_schema import (
+    validate_field_label_schema,
+)
+import fiftyone.core.fields as fof
+import fiftyone.core.labels as fol
+
+
+def get_supported_app_annotation_fields(sample_collection):
+    """
+    Gets the supported App annotation fields for a
+    :class:`fiftyone.core.collections.SampleCollection`.
+
+    Currently supported  media types for the collection are ``image`` and
+    ``3d``. See :attr:`fiftyone.core.collections.SampleCollection.media_type`
+
+    All supported primitive and ``embedded.document`` primitives are supported
+    as documented in :func:`generate_label_schema`
+
+    The below :class:`fiftyone.core.labels.Label` types are also resolved.
+
+    Supported ``image`` :class:`fiftyone.core.labels.Label` types are:
+        -   ``classification``:
+            :class:`fiftyone.core.labels.Classification`
+        -   ``classifications``:
+            :class:`fiftyone.core.labels.Classifications`
+        -   ``detection``: :class:`fiftyone.core.labels.Detection`
+        -   ``detections``: :class:`fiftyone.core.labels.Detections`
+
+    Supported ``3d`` label types are:
+        -   ``classification``:
+            :class:`fiftyone.core.labels.Classification`
+        -   ``classifications``:
+            :class:`fiftyone.core.labels.Classifications`
+        -   ``polyline``: :class:`fiftyone.core.labels.Polyline`
+        -   ``polylines``: :class:`fiftyone.core.labels.Polylines`
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+
+    Returns:
+        a list of supported fields
+    """
+    _ensure_collection_is_supported(sample_collection)
+    fields = _get_all_supported_fields(sample_collection)
+    return _flatten_fields(sample_collection, fields)
+
+
+def generate_label_schema(sample_collection, fields=None, scan_samples=True):
+    """Generates a label schema given a
```

Comment:
Same comments from `SampleCollection.generate_label_schema()` apply to this function's docstring